packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -0,0 +1,150 @@
+        maxLength: 2,
+    },
+    {
+        name: 'ratepayPhoneNumber',
+        custom: false,
+        id: 'ratepayPhoneNumber',
+        label: 'payment.ratepay.phone_number',
+        required: true,
+        fieldType: DynamicFormFieldType.TEXT,
+        maxLength: 11,
+        min: 8,
+    }
+];
+
+const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
+    method,
+    checkoutService,
+    paymentForm,
+    onUnhandledError,
+    paymentForm: {
+        isSubmitted,
+        setFieldValue,
+        setValidationSchema,
+        setSubmitted,
+    },
+    language,
+}) => {
+    const fieldsValues = useRef({});
+    const initializePayment = async () => {
+        try {
+            await checkoutService.initializePayment({
+                gatewayId: method.gateway,
+                methodId: method.id,
+                paypalcommerceratepay: {
+                    container: '#checkout-payment-continue',
+                    legalTextContainer: 'legal-text-container',
+                    getFieldsValues: () => fieldsValues.current,
+                    onRenderButton: () => {
+                        paymentForm.hidePaymentSubmitButton(method, true);
+                    },
+                    onError: (error: Error) => {
+                        paymentForm.disableSubmit(method, true);
+                        onUnhandledError(error);
+                    },
+                },
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    const deinitializePayment = async () => {
+        try {
+            await checkoutService.deinitializePayment({
+                gatewayId: method.gateway,
+                methodId: method.id,
+            });
+        } catch (error) {
+            if (error instanceof Error) {
+                onUnhandledError(error);
+            }
+        }
+    };
+
+    useEffect(() => {
+        void initializePayment();
+
+        return () => {
+            void deinitializePayment();
+        };
+    }, []);
+
+    const handleChange = useCallback(
+        (fieldId: string) => (value: string) => {
+            setFieldValue(fieldId, value);
+            updateFieldValues({
+                [fieldId]: value
+            });
+
+        },
+        [setFieldValue],
+    );
+
+    const updateFieldValues = (field: { [key: string]: unknown }) => {
+        fieldsValues.current = {...fieldsValues.current, ...field};
+    }
+
+    const validationSchema = useMemo(() =>
+            getPaypalCommerceRatePayValidationSchema({
+                formFieldData,
+                language,
+            })
+        , [language, formFieldData]);
+
+    useEffect(() => {
+        setSubmitted(false);
+        setValidationSchema(method, validationSchema);
+    }, [validationSchema, method, setValidationSchema, setSubmitted]);
+
+    return <div style={{marginBottom:'20px'}}>
```

Comment:
Why is it weird?)

Reply:
I think it's optional
Take this as advice)

Reply:
It is not weird, but in @andriiVitvitskyi1990 's case it might be better to put return code into round brackets.

```
return (
   // your code here
)
```